src/app/app.component.ts

```diff
@@ -20,9 +20,70 @@ export class AppComponent {
 
     shoppingList?.appendChild(item);
     this.itemCount += 1;
-  }
+  };
+
+  fetchAuthorizationToken = async () =>
+    await fetch('https://www.universal-tutorial.com/api/getaccesstoken', {
+      headers: new Headers({
+        'api-token': 'aJC50l2lisn1j0sD7nfyZsu0npkhf6JW2YA3FcQ1244YPtONQwiiIeU0H-6kdd6kbgY',
+        'user-email': 'abc.121009.210823@gmail.com',
+      }),
+      method: 'GET',
+      redirect: 'follow',
+    })
+      .then((response) => response.json())
+      .then(({ auth_token }) => auth_token)
+      .catch((error) => console.error(error));
+
+  fetchStates = async () => {
+    const authorizationToken = await this.fetchAuthorizationToken();
+
+    await fetch('https://www.universal-tutorial.com/api/states/United States', {
+      headers: new Headers({
+        Accept: 'application/json',
+        Authorization: `Bearer ${authorizationToken}`,
+        'api-token': 'aJC50l2lisn1j0sD7nfyZsu0npkhf6JW2YA3FcQ1244YPtONQwiiIeU0H-6kdd6kbgY',
+        'user-email': 'abc.121009.210823@gmail.com',
+      }),
+      method: 'GET',
+      redirect: 'follow',
+    })
+      .then((response) => response.json())
+      .then((states) => {
+        const abdsSelect = document.querySelector('abds-select#states');
+
+        const hiddenSelect = abdsSelect?.querySelector('select')!;
+        const hiddenSelectOptions = hiddenSelect?.querySelectorAll('option') as unknown as Node[];
+        const hiddenSelectPlaceholder = Array.from(hiddenSelectOptions).filter((option) =>
+          (option as Element).classList.contains('placeholder')
+        )[0];
+
+        if (abdsSelect) abdsSelect.innerHTML = '';
+
+        abdsSelect?.appendChild(hiddenSelect);
+
+        if (hiddenSelect) hiddenSelect.innerHTML = '';
+
+        hiddenSelect?.appendChild(hiddenSelectPlaceholder);
 
-  resetItems() {
+        states.forEach(({ state_name: state }: { state_name: string }) => {
```

Comment:
The only change i would suggest making is the way you're rendering out the options, currently it works, but it's very vanilla.... I would suggest storing the options in some kind of variable/state app.component.ts so that you can loop through in the app.component.html and render them out dynamically without having to manually set them

Reply:
I'll look into this.

Reply:
@escobarjonatan I've thought about this, and what do we gain by doing what you are suggesting? The examples are supposed to demonstrate our components so what does it matter how the data is delivered to them?

Reply:
After speaking with @escobarjonatan, I'm going to leave it as is for now and look into possibly refactoring later.